lib/rspec/mocks/method_double.rb

```diff
@@ -47,6 +47,13 @@ def object_singleton_class
         class << @object; self; end
       end
 
+      # @private
+      def method_owner
+        if Object.instance_method(:respond_to?).bind(object).call(@method_name, true)
+          Object.instance_method(:method).bind(object).call(@method_name).owner
+        end
+      end
```

Comment:
A couple things...

* It looks like this is only used from `restore_original_visibility`, below.  Can it be actually `private` instead of just being hidden from the YARD docs via `@private`?  We use `@private` only for methods that need to be publicly accessible for other parts of rspec-mocks to be able to call, but that we consider to not be part of our public API.
* What's the reason for the `instance_method(...).bind(...).call(...)` business?  (It'd probably be obvious if I had worked in rspec-mocks recently, but I haven't).  Alternately, we do have `RSpec::Support.method_handle_for(object, method_name).call(*args)` that you could use that does something similar--it might be worth using here?

Reply:
Yes, good calls, thanks.

As for `instance_method(...).bind(...).call(...)`, it's needed because the method `method` and `respond_to?` might be stubbed, and when that happened obviously it's not working.

I tried `RSpec::Support.method_handle_for` but then `spec/rspec/mocks/and_call_original_spec.rb:257` would fail with:

```
  2) and_call_original on a partial double that overrides #method works for a singleton method
     Failure/Error: RSpec::Support.method_handle_for(object, :method).call(@method_name).owner

     ArgumentError:
       wrong number of arguments (given 1, expected 0)
     # /Users/godfat/project/fork/rspec-mocks/lib/rspec/mocks/method_double.rb:56:in `method'
```

All the other tests except the still failing one is passed. Any idea? Maybe it's a bug in `RSpec::Support.method_handle_for`?